File: src/main/java/core/procedures/strings/Uppercase.java
package core.procedures.strings;

import core.procedures.AFn;
import core.procedures.FnArgsBuilder;

public final class Uppercase extends AFn {

  public Uppercase() {
    super(new FnArgsBuilder().min(1).max(1).mandatory(new Class[]{CharSequence.class}).build());
  }

  @Override
  public String getName() {
    return "upper-case";
  }

  @Override
  public String apply1(Object arg) {
    return arg.toString().toUpperCase();
  }
}

File: src/main/java/core/procedures/strings/Uppercase.kt
package core.procedures.strings

import core.procedures.AFn
import core.procedures.Arity.Exactly

class Uppercase : AFn<CharSequence?, String>(name = "upper-case", isPure = true, arity = Exactly(1),
                                             mandatoryArgsTypes = arrayOf(CharSequence::class.java)) {

    override operator fun invoke(arg: CharSequence?) = arg!!.toString().toUpperCase()
}
